src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -2172,56 +2172,68 @@ public async Task SendAsync_InvalidRequestUri_Throws()
         [InlineData('\u009F', HeaderType.Cookie)]
         public async Task SendAsync_RequestWithDangerousControlHeaderValue_ThrowsHttpRequestException(char dangerousChar, HeaderType headerType)
         {
-            string uri = "https://example.com"; // URI doesn't matter, the request should never leave the client
-            var handler = CreateHttpClientHandler();
+            TaskCompletionSource<bool> acceptConnection = new TaskCompletionSource<bool>();
 
-            var request = new HttpRequestMessage(HttpMethod.Get, uri);
-            request.Version = UseVersion;
-            try
+            await LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
```

Comment:
Nit you can disregard: this could be

```diff
-       public async Task SendAsync_RequestWithDangerous....
+       public Task SendAsync_RequestWithDangerous....
        {
            ...
-           await LoopbackServerFactory.CreateClientAndServerAsync(...
+           return LoopbackServerFactory.CreateClientAndServerAsync(...
```

Reply:
Almost all occurrences of `LoopbackServerFactory.CreateClientAndServerAsync` are with `await`, even in cases where your suggestion could be also applied. So for this reason I'll keep it as-is.